### PR TITLE
MINOR: improve exception message for incompatible Serdes to actual key/value data types

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@jcustenborder @guozhangwang @dguy @enothereska @miguno @hjafarpour

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java [hunk 1]
```diff
@@ -79,7 +79,14 @@ public void process(final K key, final V value) {
                 "Use a different TimestampExtractor to process this data.");
         }
 
-        collector.send(new ProducerRecord<K, V>(topic, null, timestamp, key, value), keySerializer, valSerializer, partitioner);
+        try {
+            collector.send(new ProducerRecord<K, V>(topic, null, timestamp, key, value), keySerializer, valSerializer, partitioner);
+        } catch (ClassCastException e) {
+            throw new StreamsException("A serializer (key: " + keySerializer.getClass().getName()
```
Comment (miguno) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java]: What about using `String#format()` to make this code more legible?

Comment (dguy) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java]: +1

**Comment by guozhangwang** [comment]
LGTM modulo @miguno 's comments.

**Comment by mjsax** [comment]
@miguno @dguy @guozhangwang Updated.

**Comment by miguno** [comment]
I think we should also include a test to verify that we are now throwing a `StreamsException` (rather than a `ClassCastException`).  I guess before we didn't have any test whatsoever though.

**Comment by mjsax** [comment]
@miguno @guozhangwang  Test added.

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**